src/BizHawk.Emulation.Cores/Consoles/Nintendo/BSNES/BsnesCore.cs

```diff
@@ -85,11 +86,18 @@ public BsnesCore(CoreLoadParameters<SnesSettings, SnesSyncSettings> loadParamete
 			InitAudio();
 			ser.Register<ISoundProvider>(_soundProvider);
 
-			if (loadParameters.Roms.Count == 2) // assume the second loaded rom is a bs memory cartridge
-			// if sufamiturbo is ever implemented this should probably do a file extension check (*.bs for bs memory)
+			if (SystemId == VSystemID.Raw.Satellaview)
 			{
-				Api.core.snes_load_cartridge_bsmemory(loadParameters.Roms[0].RomData, loadParameters.Roms[1].RomData,
-					loadParameters.Roms[0].RomData.Length, loadParameters.Roms[1].RomData.Length);
+				SATELLAVIEW_CARTRIDGE slottedCartridge = _syncSettings.SatellaviewCartridge;
+				if (slottedCartridge == SATELLAVIEW_CARTRIDGE.Autodetect)
+				{
+					slottedCartridge = (SATELLAVIEW_CARTRIDGE)loadParameters.Game.GetInt("baseCartridge", -1);
+					// TODO: this exception is kinda ugly... we could just (silently?) fallback to the BS-X cartridge as 99% of games use that one
+					if ((int)slottedCartridge == -1) throw new Exception("Unable to autodetect satellaview base cartridge for unknown game!");
```

Comment:
Using the BS-X cartridge as a fallback would be best here (that's what I was suggesting), the user can set sync settings to manually override if somehow this cartridge isn't right.

Reply:
I guess it's probably fine to do that. I just generally dislike silently failing and potentially doing unexpected things.

Reply:
You can give a warning to the user perhaps (CoreComm ShowMessage for a popup or Notify for an OSD message)